src/mitol/common/decorators.py

```diff
@@ -27,3 +30,48 @@ def _cache_controlled(request, *args, **kw):
         return _cache_controlled
 
     return _cache_controller
+
+
+def single_task(
+    timeout: int,
+    raise_block: Optional[bool] = True,
+    lock_str: Optional[str] = None,
+    cache_name: Optional[str] = "redis",
+) -> Callable:
+    """
+    Only allow one instance of a celery task to run concurrently
+    Based on https://bit.ly/2RO2aav
+
+    Args:
+        timeout(int): Time in seconds to wait before relinquishing a lock
+        raise_block(bool): If true, raise a BlockingIOError when locked
+        lock_str(str): Custom lock base name (if None, the function name will be used)
+        cache_name(str): The name of the celery redis cache (default is "redis")
+
+    Returns:
+        Callable: wrapped function (typically a celery task)
+    """
+
+    def task_run(func):
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            has_lock = False
+            client = get_redis_connection(cache_name)
+            lock_id = f"{lock_str or func.__name__}-id-{args[0] if args else 'single'}"
+            lock = client.lock(lock_id, timeout=timeout)
```

Comment:
I can't recall, so maybe you can check...does the redis client take the lock id verbatim? Wondering if there should be a `task-lock:` prefix on our lock keys so that if we ever need to find/evict them on redis it's easy to do so via a `DEL task-lock:*` command.

Reply:
I think so, I'll add the prefix

Reply:
Intended usage can be seen in this PR: https://github.com/mitodl/bootcamp-ecommerce/pull/1399/files